lm75bd/lm75bd.c

```diff
@@ -25,15 +25,64 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
   return ERR_CODE_SUCCESS;
 }
 
-error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
-  /* Implement this driver function */
-  
-  return ERR_CODE_SUCCESS;
+/*
+ * @brief Reads temperature from LM75BD
+ *
+ * @param devAddr - Address of the device to communicate to  
+ * @param temp - Float to store temperature in
+ *
+ */
+error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {  
+    float result = 0.0;
+    uint8_t targetRegister = LM75BD_POINTER_TEMP;
+    error_code_t errCode;
+    // Select sensors internal temperature register using pointer register    
+    LOG_IF_ERROR_CODE(i2cSendTo(devAddr, &targetRegister, sizeof(targetRegister)));
```

Comment:
if this fails we should just return since if we have not set the pointer register successfully, we won't be able to read the temperature. Also, since there is no return following this either, there is a chance the function continues and overwrites errCode to success somewhere else in the function and you return success even though the temp is wrong, so `RETURN_IF_ERROR_CODE` would be better